scripts/create-component/plop-templates-component/src/components/{{componentName}}/use{{componentName}}.ts.hbs

```diff
@@ -27,7 +25,7 @@ export const use{{componentName}} = (
     {
       ref: useMergedRefs(ref, React.useRef(null)),
     },
-    defaultProps,
+    defaultProps && resolveShorthandProps(defaultProps, {{camelCase componentName}}ShorthandProps),
     resolveShorthandProps(props, {{camelCase componentName}}ShorthandProps),
```

Comment:
I've been thinking about this but how much of a perf hit can it be calling `resolveShorthandProps` twice?

Reply:
It's calling it on a different set of props each time, so it's not doing extra work. I also tweaked resolveShorthandProps in this PR, to improve its perf in the case that it doesn't need to resolve anything.